lib/elixir/lib/keyword.ex

```diff
@@ -449,9 +449,20 @@ defmodule Keyword do
   """
   @spec delete(t, key, value) :: t
   def delete(keywords, key, value) when is_list(keywords) and is_atom(key) do
-    :lists.filter(fn {k, v} -> k != key or v != value end, keywords)
+    do_delete(keywords, key, value, _deleted? = false)
```

Comment:
We set out to not use `do_` in helper functions. Can we just use `delete/4` as the helper?